classes/controller/AdminController.php

```diff
@@ -3604,7 +3605,9 @@ protected function copyFromPost(&$object, $table)
                 }
```

Comment:
How about create a function `hash()` in `AdminController` ? you need it in all children classes